### PR TITLE
[10.0][ADD] New module sale_automatic_workflow_validate_purchase_mto

### PR DESCRIPTION
**Pull request by rousseldenis**
This is intended to help confirming purchase orders that are generated through sale orders directly

### CODE REVIEW

**Comment by Cedric-Pigeon** [comment]
depends on https://github.com/OCA/sale-workflow/pull/391

**Comment by lmignon** [comment]
This PR include #475

**Comment by rousseldenis** [comment]
Rebased

File: sale_automatic_workflow_validate_purchase_mto/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+{
+    'name': 'Sale Automatic Workflow Validate Purchase Mto',
+    'summary': """
+        When a sale order generates directly a purchase order,
+        validates it automatically""",
+    'version': '10.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'ACSONE SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu',
```
Comment (sergio-teruel) [inline sale_automatic_workflow_validate_purchase_mto/__manifest__.py:13]: "website": "https://odoo-community.org/" as mainteiner template

File: sale_automatic_workflow_validate_purchase_mto/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    'depends': [
+        'purchase',
+        'stock',
+        'sale_automatic_workflow'
+        ],
+    'data': [
+        'views/sale_workflow_process.xml',
+        'data/automatic_workflow.xml',
+    ],
+    'demo': [
```
Comment (sergio-teruel) [inline sale_automatic_workflow_validate_purchase_mto/__manifest__.py]: Remove empty key.

File: sale_automatic_workflow_validate_purchase_mto/data/automatic_workflow.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        <field name="user_id" ref="base.user_root"/>
+    </record>
+    <record id="automatic_workflow_po_mto_purchase_filter" model="ir.filters">
+        <field name="name">Automatic Workflow Purchase Order Purchase Filter</field>
+        <field name="model_id">purchase.order</field>
+        <field name="domain">[('state', 'not in', ['purchase', 'done','cancel'])]</field>
+        <field name="user_id" ref="base.user_root"/>
+    </record>
+    
+</odoo>
```
Comment (sergio-teruel) [inline sale_automatic_workflow_validate_purchase_mto/data/automatic_workflow.xml]: Add blank line

File: sale_automatic_workflow_validate_purchase_mto/views/sale_workflow_process.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                    <div class="oe_edit_only oe_inline">
+                        Set selection based on a search filter:
+                                        
+                        <field name="purchase_order_mto_purchase_filter_id" domain="[('model_id', '=', 'purchase.order')]" class="oe_inline" context="{'default_model_id': 'purchase.order', 'default_active': False, 'active_test': False}" can_create="true" can_write="true" />
+                    </div>
+                </div>
+            </xpath>
+        </field>
+    </record>
+
```
Comment (sergio-teruel) [inline sale_automatic_workflow_validate_purchase_mto/views/sale_workflow_process.xml]: Remove blank lines.

File: sale_automatic_workflow_validate_purchase_mto/__manifest__.py (hunk 1 above)
Comment (sergio-teruel) [inline sale_automatic_workflow_validate_purchase_mto/__manifest__.py:13]: Add "website": "https://odoo-community.org/"

File: sale_automatic_workflow_validate_purchase_mto/__manifest__.py (hunk 2 above)
Comment (sergio-teruel) [inline sale_automatic_workflow_validate_purchase_mto/__manifest__.py]: Remove empty key

File: sale_automatic_workflow_validate_purchase_mto/data/automatic_workflow.xml (hunk 3 above)
Comment (sergio-teruel) [inline sale_automatic_workflow_validate_purchase_mto/data/automatic_workflow.xml]: Add blank line

File: sale_automatic_workflow_validate_purchase_mto/views/sale_workflow_process.xml (hunk 4 above)
Comment (sergio-teruel) [inline sale_automatic_workflow_validate_purchase_mto/views/sale_workflow_process.xml]: Remove blank lines

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: sale_automatic_workflow_validate_purchase_mto/views/sale_workflow_process.xml (hunk 4 above)
Comment (sergio-teruel) [reply sale_automatic_workflow_validate_purchase_mto/views/sale_workflow_process.xml]: Functional test Ok.

File: sale_automatic_workflow_validate_purchase_mto/__manifest__.py (hunk 1 above)
Comment (rousseldenis) [reply sale_automatic_workflow_validate_purchase_mto/__manifest__.py:13]: @sergio-teruel It's not mandatory in OCA guidelines

https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules

**Comment by rousseldenis** [comment]
Fixed and squashed

**Review by sergio-teruel — APPROVED** [review]

**Review by Cedric-Pigeon — APPROVED** [review]

File: sale_automatic_workflow_validate_purchase_mto/__manifest__.py (hunk 1 above)
Comment (sbidoul) [reply sale_automatic_workflow_validate_purchase_mto/__manifest__.py:13]: @rousseldenis please https://github.com/OCA/sale-workflow .

**Comment by rousseldenis** [comment]
@OCA/crm-sales-marketing-maintainers

**Comment by rafaelbn** [comment]
Hello @rousseldenis , travis failed when merging... please could you take a look?

> 2019-10-01 23:50:16,263 6010 ERROR openerp_test odoo.addons.sale_delivery_split_date.tests.test_sale_delivery: FAIL

https://travis-ci.org/OCA/sale-workflow/jobs/592176090

Thanks! Let me know

**Comment by rousseldenis** [comment]
@rafaelbn I've rebased this one. Let's see

**Comment by rafaelbn** [comment]
@rousseldenis travis is :red_circle: due LINT errors

`sale_automatic_workflow_validate_purchase_mto/readme/CONFIGURATION.rst:4: [E7901(rst-syntax-error), ]  Bullet list ends without a blank line; unexpected unindent.`

**Comment by rousseldenis** [comment]
@rafaelbn

**Pull request merged**